shell/browser/native_window_views.cc

```diff
@@ -462,9 +446,17 @@ NativeWindowViews::~NativeWindowViews() {
   SetForwardMouseMessages(false);
 #endif
 
-  aura::Window* window = GetNativeWindow();
-  if (window)
+  if (aura::Window* const window = GetNativeWindow())
     window->RemovePreTargetHandler(this);
+
+  if (is_modal()) {
+    if (NativeWindow* const parent = this->parent()) {
+      // Enable parent window after current window gets closed.
+      static_cast<NativeWindowViews*>(parent)->DecrementChildModals();
+      // Focus on parent window.
+      parent->Focus(true);
+    }
```

Comment:
Is there a case when `OnWidgetDestroying` won't be called, why duplicate it in the delegate destructor ?

Reply:
Oof, the destructor code was from a different draft that wasn't supposed to be in this PR. :sweat_smile: 

I've force-pushed a cleanup s.t. everything is handled in `OnWidgetDestroying()` and `OnWidgetDestroyed()`.